js/ui/calendar/ui.calendar.js

```diff
@@ -243,28 +256,36 @@ const Calendar = Editor.inherit({
         return dateSerialization.getDateSerializationFormat(value);
     },
 
-    _convertToDate: function(value, optionName) {
+    _convertToDate: function(value) {
         return dateSerialization.deserializeDate(value);
     },
 
     _dateValue: function(value, event) {
+        const optionName = Array.isArray(value) ? 'values' : 'value';
         if(event) {
             if(event.type === 'keydown') {
                 const cellElement = this._view._getContouredCell().get(0);
                 event.target = cellElement;
             }
             this._saveValueChangeEvent(event);
         }
-        this._dateOption('value', value);
+        this._dateOption(optionName, value);
     },
 
     _dateOption: function(optionName, optionValue) {
         if(arguments.length === 1) {
-            return this._convertToDate(this.option(optionName), optionName);
+            const values = this.option('values') ?? [];
+            return optionName === 'values'
```

Comment:
Looks like this condition and others should be processed on the 'strategy' level. What do you think, it's possible?

Reply:
It's sure possible but requeires to rework this function, cause it's too "wide" now.  For example it's also used to get `min/max` dates, so we will also need to check if optionName is `values` or not on multiselect strategies. Also it's used in `_updateTimeComponent` function that set hours/minutes/seconds of current date depending on `value`.
Without refactoring this function on strategies will be still pretty ugly, so I desided to not move it there